lib/rules/no-else-return.js

```diff
@@ -88,6 +102,34 @@ module.exports = {
                     }
 
                     const endToken = sourceCode.getLastToken(node);
+
+                    /**
+                     * If else block includes block scope local variable declaration [let or const],
+                     * then it is not safe to remove else keyword [issue 11069]
+                     */
+                    const isDeclarationInside = sourceCode.getTokensBetween(startToken, endToken).findIndex(isVariableDeclaration) > -1;
+
+                    const childScopeVars = (childScopes, store = {}, level = 0) =>
+                        childScopes.reduce((acc, block) => {
+                            if (block.childScopes.length >= 0) {
+                                acc[level] = block.variables;
+                                let index = level + 1
+                                return childScopeVars(block.childScopes, acc, index);
+                            }
+                            return acc
+                        }, store)
+
+                    const shadowedVariables = scope => {
+                        const result = childScopeVars(scope.childScopes);
+                        return scope.variables.filter(elem => result[0].findIndex(vars => vars.name === elem.name) > -1)
+                    };
+                    const shadowed = shadowedVariables(context.getScope());
```

Comment:
you probably don't need to compute the shadowed names. Consider the following example, where the shadowed declaration is in a parent scope:

```js
let foo = 1;
function test() {
  if (condition) {
    return foo;
  } else {
    let foo = 2;
    console.log(foo)
  }
}
```

removing the `else`-block changes the reference of `foo` in the `then`-branch (which will throw as the use is in the temporal dead zone)

Reply:
@ajafff Thank you very much for your example and patience. I made another one commit. Can you check it?
Still have problem with variable shadowing,Don't know how to handle it.
P.S. Happy holidays!